k4FWCore/python/k4FWCore/utils.py

```diff
@@ -18,10 +18,37 @@
 # limitations under the License.
 #
 import os
+import re
 from io import TextIOWrapper
 from typing import Union
 
 
+def check_wrong_imports(code: str) -> None:
+    """Check for wrong imports in the given code.
+
+    This function checks the given code for any imports of IOSvc or ApplicationMgr
+    from Configurables instead of k4FWCore. If such an import is found, an ImportError
+    is raised.
+
+    Args:
+        code (str): The code to check for wrong imports.
+
+    Raises:
+        ImportError: If the code contains an import of IOSvc or ApplicationMgr from Configurables.
+
+    """
+    # Check first that IOSvc is being used, in that case
+    # Importing either ApplicationMgr or IOSvc from Configurables is not allowed
+    iosvc_regex = re.compile(
+        r"^\s*from\s+(Configurables|k4FWCore)\s+import\s+\(?.*IOSvc.*\)?", re.MULTILINE
+    )
+    regex = re.compile(
+        r"^\s*from\s+Configurables\s+import\s+\(?.*(ApplicationMgr|IOSvc).*\)?", re.MULTILINE
+    )
+    if re.search(iosvc_regex, code) and re.search(regex, code):
```

Comment:
As discussed in our Key4hep meeting, it's not easy to make a comprehensive set of checks and the checks in place are put there to catch probably most occurrences since they will typically be the simplest case, `from Configurables import ...`. I have added some comments about how the checks won't catch every case.